common/archiver/s3store/historyArchiver.go

```diff
@@ -303,7 +303,7 @@ func (h *historyArchiver) Get(
 				return nil, &types.InternalServiceError{Message: err.Error()}
 			}
 			switch err.(type) {
-			case *types.BadRequestError, *types.InternalServiceError, *types.EntityNotExistsError:
+			case *types.BadRequestError, *types.InternalServiceError, *types.EntityNotExistsError, *types.WorkflowExecutionAlreadyCompletedError:
```

Comment:
I don't think already completed error will be returned here by the download func?